tests/unit/introspection.tcl

```diff
@@ -54,6 +54,30 @@ start_server {tags {"introspection"}} {
         # After killing `me`, the first ping will throw an error
         assert_error "*I/O error*" {r ping}
         assert_equal "PONG" [r ping]
+
+        $rd1 close
+        $rd2 close
+        $rd3 close
+        $rd4 close
+    }
+
+    foreach bg_cmd {bgsave bgrewriteaof} {
+        test "CLIENT KILL close the client connection during $bg_cmd" {
+            # Start a slow bgsave or bgrewriteaof, trigger an active fork.
+            r set k v
+            r config set rdb-key-save-delay 10000000
+            r $bg_cmd
+
+            # Kill (close) the connection
+            r client kill skipme no
+
+            # In the past, client connections needed to wait for bgsave or bgrewriteaof
+            # to end before actually closing, now they are closed immediately.
+            assert_error "*I/O error*" {r ping} ;# get the error very quickly
+            assert_equal "PONG" [r ping]
+
+            r config set rdb-key-save-delay 0
+        } {OK}
```

Comment:
there's a problem here.
the sleep is for 10 seconds (supposedly we have one key, but actually maybe we better add FLUSHALL)
but then, we continue to the next test (a loop) without terminating the bgsave, so the AOFRW command is just getting scheduled.

what i suggest:
1. add a flushall
2. add an assertion after the client disconnected to make sure `rdb_bgsave_in_progress`
3. stop the child before we proceed to the next test. (we can do that with FLUSHALL, and then for fun, add a wait_for check that it's no longer running)
4. i don't think we need to test BGREWRITEAOF too.

Reply:
wow, thanks, I didn't think about this scheduled thing (i did thought about adding a flag check to make sure the child is alive before). the suggestion is great, very complete and detailed, and i dropped the AOFRW, thanks again